crates/biome_js_analyze/src/lint/nursery/no_destructured_props.rs

```diff
@@ -159,11 +159,16 @@ impl Rule for NoDestructuredProps {
     }
 }
 
-fn is_inside_jsx_component(binding_pattern: &JsObjectBindingPattern) -> Option<bool> {
-    let variable_declarator = binding_pattern
+fn is_inside_jsx_component(parameters: &JsParameters) -> Option<bool> {
+    let arrow_function_expression = parameters
         .syntax()
-        .ancestors()
-        .find_map(JsVariableDeclarator::cast)?;
+        .parent()
+        .and_then(JsArrowFunctionExpression::cast)?;
```

Comment:
How about function declarations?

```js
function Component({ foo }: Props) {
  // ...
}
```

Reply:
Strangely, the testing suite of the eslint rule doesn't test function declarations or function expressions.

I'm not a SolidJS user, so I don't know if it's expected or not